sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventDataBatch.java

```diff
@@ -81,19 +81,16 @@ public int getSizeInBytes() {
      */
     public boolean tryAdd(final EventData eventData) {
         if (eventData == null) {
-            logger.logAndThrow(new IllegalArgumentException("eventData cannot be null"));
-            return false;
+            throw logger.logWarningAndThrow(new IllegalArgumentException("eventData cannot be null"));
         }
 
         final int size;
         try {
             size = getSize(eventData, events.isEmpty());
         } catch (java.nio.BufferOverflowException exception) {
```

Comment:
This is out of scope of this PR but while you are modifying this file, could you also instead import the package and remove this fully-qualified class name?